src/main/java/org/jvnet/solaris/libzfs/ZFSObject.java

```diff
@@ -199,17 +206,29 @@ public ZFSSnapshot createSnapshot(final String snapshotName) {
     public ZFSSnapshot createSnapshot(final String snapshotName,
             final boolean recursive) {
         String fullName = name + '@' + snapshotName;
-        String abi = library.getFeature("LIBZFS4J_ABI_zfs_snapshot");
+        String abi_thisfunc = "createSnapshot";
+        String abi_toggle = "LIBZFS4J_ABI_zfs_snapshot";
+        String abi = library.getFeature(abi_toggle);
+        if (abi.equals("NO-OP")) {
+            LOGGER.log(Level.FINE, "NO-OP: libzfs4j::" + abi_thisfunc + "() was called while " + abi_toggle + "=='" + abi + "' - skipped due to config");
+        } else
+        if (abi.equals("openzfs") || abi.equals("legacy")) {
+            /* good for both "openzfs" and "legacy" as we know them today */
+            if (LIBZFS.zfs_snapshot(library.getHandle(), fullName, recursive, null) != 0) {
+                throw new ZFSException(library);
+            }
+        } else
         if (abi.equals("pre-nv96")) {
             /* Very-very old, prehistoric signature */
+            /* Be careful to not call this signature on newer OSes though,
+             * because it is a subset of newer ABI and so does not cause a
+             * link error - but may provide random data as the last arg. */
             if (LIBZFS.zfs_snapshot(library.getHandle(), fullName, recursive) != 0) {
                 throw new ZFSException(library);
             }
         } else {
-            /* good for both "openzfs" and "legacy" as we know them today */
-            if (LIBZFS.zfs_snapshot(library.getHandle(), fullName, recursive, null) != 0) {
-                throw new ZFSException(library);
-            }
+            LOGGER.log(Level.FINE, "NO-OP: libzfs4j::" + abi_thisfunc + "() was called while " + abi_toggle + "=='" + abi + "' and this is currently not a known value");
+            throw new ZFSException(library);
```

Comment:
Well, this toggle never (explicitly) supported that value used just in a couple of other functions. Now that would be explicitly rejected, as we had no good research into that (and also because "before update 8" there were a number of releases to think about in theory, but less so in practice).

So sort of "as designed" - handles the big-toggle settings.